wrapanapi/systems/msazure.py

```diff
@@ -1077,6 +1077,42 @@ def list_compute_images(self):
         return self.resource_client.resources.list(
             filter="resourceType eq 'Microsoft.Compute/images'")
 
+    def list_compute_images_by_resource_group(self, resource_group=None, free_images=None):
+        """
+        Args:
+            resource_group (str): Name of the resource group
+            free_images (bool): Whether to collect image which do not have any resource(VM) linked to it
+        """
+        resource_group = resource_group or self.resource_group
+        image_list = list(
+            self.resource_client.resources.list(
+                filter=f"resourceType eq 'Microsoft.Compute/images' and resourceGroup eq '{resource_group}'"
+            )
+        )
```

Comment:
I am not really in favour of changing the working of the existing list_compute_images() as it might affect the working wherever this is being used.

Reply:
It would be less duplication, but @akhil-jha is following a consistent pattern here, there are several methods in the module that follow this pattern.